chesster.js

```diff
@@ -636,115 +522,63 @@ controller.hears([
     });
 });
 
-controller.hears([
-    players.appendPlayerRegex("pairing", true)
-], [
-    'direct_mention',
-    'direct_message'
-], function(bot, message) {
-    bot_exception_handler(bot, message, function() {
-        bot.startPrivateConversation(message, function (response, convo) {
-            var requestingPlayer = users.getByNameOrID(message.user);
-            var player = players.getSlackUser(users, message);
-
-            preparePairingCompetitorMessage(requestingPlayer, player, function(response) {
-                convo.say(response);
-            });
+slack.hears(controller, {
+    patterns: [
+        players.appendPlayerRegex("pairing", true)
+    ],
+    message_types: [
+        'direct_mention', 'direct_message'
+    ],
+    config: config,
+}, function(bot, message) {
+    var targetPlayer = players.getSlackUser(users, message);
+    var deferred = Q.defer();
+    var allLeagues = league.getAllLeagues(config);
+    bot.startPrivateConversation(message, function (response, convo) {
+        Q.all(
```

Comment:
As I understand it, `Q.all(...)` is meant to return an array of results. Are we using the array result somewhere?

Do we need an array result that I dont see?

If not, is this the right method for our use case?